src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.Utf8JsonStreamReader.cs

```diff
@@ -1169,6 +1173,32 @@ private static void ReadMSBuildMetadata(ref Utf8JsonStreamReader jsonReader, Pac
 
                         warningProperties = new WarningProperties(warnAsError, noWarn, allWarningsAsErrors, warningsNotAsErrors);
                     }
+                    else if (jsonReader.ValueTextEquals(UsingMicrosoftNETSdk))
+                    {
+                        usingMicrosoftNetSdk = jsonReader.ReadNextTokenAsBoolOrTrue(Encoding.UTF8.GetString(UsingMicrosoftNETSdk));
```

Comment:
`Encoding.UTF8.GetString` will cause a heap allocation, and restore is our most performance critical code, so we must eliminate all unnecessary allocations.

One option is to pass the `byte[]` or `Span<byte>` to the `ReadNextTokenAsBoolOrTrue`, and only convert it to a string when the string is needed (in the error handling code), so it's not allocated in the non-error code path.

However, knowing a bit more about how .NET works, you can create a const string, which you pass to `Encoding.UTF8.GetBytes` where the `static readonly byte[]` is defined, because that `byte[]` is calculated at runtime, which means that if you inspect the compiled assembly, there's already a `"UsingMicrosoftNETSdk"` const string. Therefore, creating another const and using it here, instead of `UTF8.GetString` will avoid the allocation, and it won't result in any new additional string constants in the assembly.  However, since the variable name is identical, you can also use `nameof(UsingMicrosoftNETSdk)`.

So, to reiterate, anything related to restore, make sure you keep high performance code in mind, and avoid unnecessary heap allocations. There are several ways to achieve it, so you have flexibility on how you do it.